pkg/binding/bindingpolicy-resolver.go

```diff
@@ -135,6 +125,43 @@ type BindingPolicyResolver interface {
 	Broker() ResolutionBroker
 }
 
+// DownsyncModulation is a convenient internal representation of v1alpha1.DownsyncModulation
+type DownsyncModulation struct {
+	CreateOnly                 bool
+	StatusCollectors           sets.Set[string]
+	WantSingletonReportedState bool
+}
+
+func ZeroDownsyncModulation() DownsyncModulation {
+	return DownsyncModulation{StatusCollectors: sets.New[string]()}
+}
+
+func DownsyncModulationFromExternal(external v1alpha1.DownsyncModulation) DownsyncModulation {
+	return DownsyncModulation{
+		CreateOnly:                 external.CreateOnly,
+		StatusCollectors:           sets.New(external.StatusCollectors...),
+		WantSingletonReportedState: external.WantSingletonReportedState,
+	}
+}
+
+func (dm *DownsyncModulation) ToExternal() v1alpha1.DownsyncModulation {
+	return v1alpha1.DownsyncModulation{
+		CreateOnly:                 dm.CreateOnly,
+		StatusCollectors:           sets.List(dm.StatusCollectors),
+		WantSingletonReportedState: dm.WantSingletonReportedState,
+	}
+}
+
+func (left *DownsyncModulation) Equal(right DownsyncModulation) bool {
+	return left.CreateOnly == right.CreateOnly && left.WantSingletonReportedState == right.WantSingletonReportedState && left.StatusCollectors.Equal(right.StatusCollectors)
+}
+
+func (dm *DownsyncModulation) AddExternal(external v1alpha1.DownsyncModulation) {
```

Comment:
I can't find any use of this function in this repo. I also wonder when would this be useful? A modulation is a member of a resolution, and a resolution is resolved from a single binding-policy.